### PR TITLE
[FEAT] Goblin Town POC

### PR DESCRIPTION
**Pull request by adamhannigan**
# Description

This PR introduces the concept of Goblin town and cooperates with the new smart contract upgrades. In this version, more data (such as crafting recipes) now live completely on chain.

This moves us closer to our vision of decentralisation 🚀 

It is currently still in draft and will be worked on over the coming days.

**Includes**

- Town Hall
- Goblin Town
- React Router
- Wishing Well Updates

## Type of change

Please delete options that are not relevant.

- [X] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Manually tested against local API

# Checklist:

- [X] Title of the PR is relevant and is prefixed with [FEAT], [CHORE] or [FIX]
- [X] I have read the contributing guidelines and agree to the T&Cs

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -4,6 +4,14 @@ import { Navigation } from "./Navigation";
 
 import "./styles.css";
 
+import { inspect } from "@xstate/inspect";
+
+inspect({
```
Comment (adamhannigan) [inline src/App.tsx:9]: @spencerdezartsmith Do we need to wrap this in a dev flag?

**Comment by Pranav-yadav** [comment]
Something special is cooking.. 🎉 
I just can't stop myself from spamming this PR discussions... 🙌 

Btw, will review if I get time..
Thanks @adamhannigan @spencerdezartsmith and @Rofrtd, this is awesome :heart:

Side Note: Guys use `[skip ci]` in commit messages whenever possible... Like for layout fixes, touchups 🤗 🌏

**Comment by Pranav-yadav** [comment]
Oops, forgot to mention that I've been also cooking something special since this morning.... 🤫 

Maybe, Wallet 👀  😜
Will open the WIP PR... 😉

**Comment by adamhannigan** [comment]
@Pranav-yadav Great tip!

The aim is to get this into develop ASAP to get the army of open source developers contributing on this as it will require a huge team of helpers ❤️

**Pull request merged**